mock-thunderhead/src/main/resources/application.yml

```diff
@@ -70,6 +70,7 @@ auth:
     azure:
       disk:
         SSEWithCMK.enable: true
+      encryptionAtHost.enable: true
```

Comment:
Sanity check:
* Is Encryption at Host enabled in the `azure-eng-cloud-daily` subscription? (My assumption is it is, otherwise you would have no way to verify the feature.)
* Does it have any foreseeable negative effects if this gets enabled for all the Azure deployments (when using mock-thunderhead)? Will Encryption at Host need to be explicitly requested during environment launch, or it will rather be automatically applied whenever the entitlement is granted for the tenant? (In other words, will customers have a choice to opt out not to use Encryption at Host?)

Reply:
Reference - [enable encryptionAtHost feature for subscription](https://docs.microsoft.com/en-us/azure/virtual-machines/disks-enable-host-based-encryption-portal#prerequisites)

- I registered/un-registered the feature for `azure-eng-cloud-daily` to test and try. Without enabling the feature Azure does not recognize "encryptionAtHost" variable in templates.

- Enabling CDP entitlement means that customer has performed the pre-requisite of enabling feature in their subscription. I'll make sure this is documented correctly. 

- I see, if we `enable: true` this in mock-thunderhead - and if mock-thunderhead is used for tests in other subscriptions, all other subscriptions should have this enabled! 
hmm, may be I should set it to false then and should enable for my local testing only.

Reply:
> Enabling CDP entitlement means that customer has performed the pre-requisite of enabling feature in their subscription. I'll make sure this is documented correctly.

You mean, if they want to use it in some subscription, there is no way not to use it in other subscriptions? If they have the entitlement, deployments in subscriptions where the feature is not enabled / registered will fail, if I interpreted your response correctly. :frowning:  (Or what exactly is the consequence of "Azure does not recognize" the `encryptionAtHost` in the ARM deployment template? Will it break the deployment, or simply `encryptionAtHost` will be silently ignored?)

> `enable: true` in `mock-thunderhead`

So, depending on the answer to my above question in the previous paragraph, if the deployment breaks when `encryptionAtHost` is present in the ARM deployment template, no one will be able to launch Azure environments unless the target subscription has the feature enabled. This impacts local dev as well as E2E nightly tests.

Reply:
> If they have the entitlement, deployments in subscriptions where the feature is not enabled / registered this change will fail.

Yes. it will break the deployment.
As per your suggestions above and my understanding, I am setting `enable: false` in mock-thunderhead

Reply:
I have disabled the entitlement in `mock-thunderhead`. cc: @lajosrodek

Reply:
Thanks. Circling back to my first point: Is it acceptable that, whenever the entitlement is present for a given CDP tenant, all the Azure subscriptions involved in any CDP environments within that tenant must then have Encryption at Host enabled?

That is, given the customer has SubscriptionA and SubscriptionB, has the CDP entitlement, but only has Encryption at Host enabled for SubscriptionA, they will not be able to launch environments into SubscriptionB (because the ARM deployment will fail as you noted).

CC @shwethags Would be interested in your opinion as well. I would rather expect this feature be an "opt in" thing, i.e. something explicitly asked during environment launch with a dedicated CLI option / UI checkbox.